### PR TITLE
fix(Badge): fix accidentally rendering an empty element when text props is empty

### LINKED ISSUE #37678: Badge component does not pass text props, there is an empty elemet with a left margin
### Reproduction link

[Edit on CodeSandbox](https://codesandbox.io/s/ji-ben-yong-fa-antd-4-23-2-forked-ttqe0l?file=/demo.js:210-215)

### Steps to reproduce

image

### What is expected?

cancel left margin when empty or not render element

### What is actually happening?

renders an empty element with a left margin

| Environment | Info |
| --- | --- |
| antd | 4.23.2 |
| React | all |
| System | all |
| Browser | all |

### PR DESCRIPTION
**Pull request by Wxh16144**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

close #37678

### 💡 Background and solution

Determine not to render a node if the text property is empty.

### 📝 Changelog

| Language   | Changelog                                                                                   |
| ---------- | ------------------------------------------------------------------------------------------- |
| 🇺🇸 English | Fix the problem of rendering an empty element with styles when Badge not pass text property |
| 🇨🇳 Chinese | 修复 Badge 未传入 text 属性时，渲染了一个带有样式的空元素问题                               |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/badge/index.tsx [hunk 1]
```diff
@@ -165,9 +165,11 @@ const Badge: CompoundedComponent = ({
...
       <span {...restProps} className={badgeClassName} style={mergedStyle}>
         <span className={statusCls} style={statusStyle} />
-        <span style={{ color: statusTextColor }} className={`${prefixCls}-status-text`}>
-          {text}
-        </span>
+        {text && (
+          <span style={{ color: statusTextColor }} className={`${prefixCls}-status-text`}>
+            {text}
+          </span>
+        )}
```
Comment (Wxh16144) [inline components/badge/index.tsx:172]: 这里的格式是否需要美化一下？

```jsx
{
  text && (
    <span
      style={{ color: statusTextColor }}
      className={`${prefixCls}-status-text`}
    >
      {text}
    </span>
  )
}
```

File: components/badge/demo/colorful.md [hunk 2]
```diff
@@ -36,31 +36,20 @@ const colors = [
...
-      <Badge color="#2db7f5" text="#2db7f5" />
-      <br />
-      <Badge color="#87d068" text="#87d068" />
-      <br />
-      <Badge color="#108ee9" text="#108ee9" />
-    </>
+      <Badge color="rgb(45, 183, 245)" text="rgb(45, 183, 245)" />
+      <Badge color="hsl(102, 53%, 61%)" text="hsl(102, 53%, 61%)" />
+      <Badge color="hwb(205 6% 9%)" text="hwb(205 6% 9%)" />
+    </Space>
```
Comment (Wxh16144) [inline components/badge/demo/colorful.md:50]: 可以直观的向开发者描述 `<Badge color="xxx" />` 支持多种描述颜色方式。[How to describe a color](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Colors/Applying_color#how_to_describe_a_color)

File: components/badge/index.tsx (hunk 1 above)
Comment (afc163) [reply components/badge/index.tsx:172]: 可以，不过一般按 precommit 自动格式化的效果就好。

Comment (Wxh16144) [reply components/badge/index.tsx:172]: > 可以，不过一般按 precommit 自动格式化的效果就好。

precommit 没有对这次提交做任何格式化操作

Comment (afc163) [reply components/badge/index.tsx:172]: 重装一下 node_modules ？

Comment (Wxh16144) [reply components/badge/index.tsx:172]: 没问题，我尝试一下

Comment (MadCcc) [reply components/badge/index.tsx:172]: `{}` 里的代码没有换行的规则。。我个人喜欢不换行

Comment (Wxh16144) [reply components/badge/index.tsx:172]: > `{}` 里的代码没有换行的规则。。我个人喜欢不换行

嗯哼🤔，那我就不做处理咯

**Comment by MadCcc** [comment]
changelog 记得填一下

**Comment by afc163** [comment]
ci 挂了

**Comment by afc163** [comment]
还是挂的

**Comment by Wxh16144** [comment]
> 还是挂的

之前被 `Form › Form.Item.useStatus should work` 这个用例卡了，没找到问题，看到前面大佬已经修复了。我重新 rebase 再跑了一遍 case

**Pull request merged**